src/repositories/soft-crud.repository.base.ts

```diff
@@ -106,6 +111,45 @@ export abstract class SoftCrudRepository<
     return super.findOne(filter, options);
   }
 
+  //findOne() for soft deleted entry
+  findOneWithSoftDelete(
+    filter?: Filter<T>,
+    options?: Options,
+  ): Promise<(T & Relations) | null> {
+    // Filter out soft deleted entries
+    if (
```

Comment:
no need of this logic. Just trigger the original findOne function inside this. That will do find on soft deleted records as well.

Reply:
update of the readme is done in the latest commit and make changes in the findOneIncludeSoftDelete function as well.